### PR TITLE
[esp/bindings_js] load scene files bases on URL parameters

### PR DESCRIPTION
**Pull request by msbaines**
Instead of using a large .data file, we preload the scene files
into the virtual filesystem based on URL scene parameter. This is
a workaround until Corrade PR#39 is ready.

This should speed up startup since we now only load the files we
need.

## Motivation and Context

Speedup startup by only loading files we need for the current scene.

## How Has This Been Tested

Verified that the new URL parameter scene works.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ X I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/esp/bindings_js/index.js [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
 import VRDemo from "./modules/vr_demo";
 import "./bindings.css";
 
-Module["onRuntimeInitialized"] = function() {
-  console.log("hsim_bindings initialized");
+Module.preRun.push(() => {
+  let arg;
+  let scene;
+  for (arg of window.location.search.substr(1).split("&")) {
+    let [key, value] = arg.split("=");
```
Comment (apsdehal) [inline src/esp/bindings_js/index.js]: I would add a default value here and make sure that args has length 2 before assigning it to `key, value`.

File: src/esp/bindings_js/index.js [hunk 2]
```diff
@@ -1,12 +1,26 @@
...
+    if (key == "scene") {
```
Comment (apsdehal) [inline src/esp/bindings_js/index.js]:
```suggestion
    if (key === "scene") {
```

File: src/esp/bindings_js/index.js [hunk 3]
```diff
@@ -1,12 +1,26 @@
...
+      scene = value;
+    }
+  }
+  FS.createPreloadedFile("/", scene, scene, true, false);
+  Module.scene = scene;
+  let navmesh = scene.replace(/\..*/, ".navmesh");
```
Comment (apsdehal) [inline src/esp/bindings_js/index.js]: To cover all kind of cases
```suggestion
  const navmesh = scene.substr(0, scene.lastIndexOf(".")) + ".navmesh";
```

File: src/esp/bindings_js/index.js [hunk 4]
```diff
@@ -1,12 +1,26 @@
...
 
 import WebDemo from "./modules/web_demo";
 import VRDemo from "./modules/vr_demo";
 import "./bindings.css";
 
-Module["onRuntimeInitialized"] = function() {
-  console.log("hsim_bindings initialized");
+Module.preRun.push(() => {
+  let arg;
+  let scene;
```
Comment (apsdehal) [inline src/esp/bindings_js/index.js]: NIT: To make it clear that this is url, maybe name it sceneUrl?

File: src/esp/bindings_js/index.js [hunk 5]
```diff
@@ -1,12 +1,26 @@
...
+  for (arg of window.location.search.substr(1).split("&")) {
+    let [key, value] = arg.split("=");
+    if (key == "scene") {
+      scene = value;
+    }
+  }
+  FS.createPreloadedFile("/", scene, scene, true, false);
```
Comment (apsdehal) [inline src/esp/bindings_js/index.js]: Does this cache it once it is loaded?

Comment (msbaines) [reply src/esp/bindings_js/index.js]: In chrome inspector, I see that it is served from the disk cache.

File: src/esp/bindings_js/index.js (hunk 4 above)
Comment (msbaines) [reply src/esp/bindings_js/index.js]: Its also the filename we are using in the virtual filesystem.

File: src/esp/bindings_js/index.js (hunk 1 above)
Comment (msbaines) [reply src/esp/bindings_js/index.js]: Good idea! That also preserves the prior behavior.

**Pull request merged**